File: How_To_Program_Exercises/C++How_To_Program_8/Chapter3/InvoiceClass/InvoiceTest.cpp
//
//	InvoiceTest: Tests the Invoice class
//

#include <iostream>
#include <string>
#include "Invoice.h"

int main ()
{
	std::cout << "Creating first invoice\n-----------------------\n";
	Invoice invoice1 (40, 4, "12123", "Flaming Star");
	invoice1.setDescription("The Flaming Star");
	std::cout << "\nItem description: " << invoice1.getDescription()
			  << "Item number: " << invoice1.getNumber() 
			  << "\nItem price: " << invoice1.getPrice()
			  << "\nItem quantity: " << invoice1.getQuantity()
			  << "\nInvoice total: " << invoice1.getInvoice() 
			  << std::endl;
	
	std::cout << "\n\n\nCreating second invoice\n-------------------------\n";
	Invoice invoice2 (25, 11, "44344", "Element of Surprise");
	std::cout << "\nItem description: " << invoice2.getDescription()
		<< "Item number: " << invoice2.getNumber() 
		<< "\nItem price: " << invoice2.getPrice()
		<< "\nItem quantity: " << invoice2.getQuantity()
		<< "\nInvoice total: " << invoice2.getInvoice()
		<< std::endl;
	
	
	std::cout << "\n\n\nCreating third invoice\n-----------------------\n";
	Invoice invoice3 (-12, -1, "24994", "Ready to operate");

	std::cout << "\nItem description: " << invoice3.getDescription() 
		<< "Item number: " << invoice3.getNumber()
		<< "\nItem price: " << invoice3.getPrice()
		<< "\nItem quantity: " << invoice3.getQuantity()
		<< "\nInvoice total: " << invoice3.getInvoice()
		<< std::endl;

	invoice3.setPrice(120);
	invoice3.setQuantity(2);
	std::cout << "\n\nAfter setting correct quantity and price for invoice3:\n";
	std::cout << "\nItem description: " << invoice3.getDescription()
		<< "Item number: " << invoice3.getNumber()
		<< "\nItem price: " << invoice3.getPrice()
		<< "\nItem quantity: " << invoice3.getQuantity()
		<< "\nInvoice total: " << invoice3.getInvoice()
		<< std::endl;

}

File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/InvoiceClass/InvoiceTest.java
//
//	Invoice Test Class
//

public class InvoiceTest
{
	public static void main (String[] args)
	{
		System.out.println("Creating the first invoice\n---------------------------");
		Invoice invoice1 = new Invoice ("95832", "Rick And Morty Season 1", 3, 24.99);
		System.out.printf("Item Description: %s\nItem number: %s\nItem unit price: $%.2f\nItem quantity: %d\nInvoice: $%.2f\n",
						  invoice1.getDescription(), invoice1.getNumber(), invoice1.getPricePerItem(),
						  invoice1.getQuantity(), invoice1.getInvoiceAmount());
		
		System.out.println("\n\n\nCreating the second invoice with negative quantity and price\n------------------------------------------------------------");
		Invoice invoice2 = new Invoice ("A", "B", -4, -20.98);
		
		System.out.printf("Item Description: %s\nItem number: %s\nItem unit price: $%.2f\nItem quantity: %d\nInvoice: $%.2f\n",
						  invoice2.getDescription(), invoice2.getNumber(), invoice2.getPricePerItem(),
						  invoice2.getQuantity(), invoice2.getInvoiceAmount());
						  
		invoice2.setNumber ("44343");
		invoice2.setDescription ("The Zoidberg Tragedy");
		invoice2.setPricePerItem (19.98);
		invoice2.setQuantity (5);
		System.out.println("\n\nAfter setting fields to correct values\n---------------------------------------");
		System.out.printf("Item Description: %s\nItem number: %s\nItem unit price: $%.2f\nItem quantity: %d\nInvoice: $%.2f\n",
						  invoice2.getDescription(), invoice2.getNumber(), invoice2.getPricePerItem(),
						  invoice2.getQuantity(), invoice2.getInvoiceAmount());
		
	}
}